### ISSUE TITLE
Add cookie consent notice blocking P3A metrics

### ISSUE THREAD

**Issue by DJAndries** [brave/brave-browser#25568]
## Description 

Add two P3A questions for cookie consent notice blocking settings.

"If you have viewed the cookie consent block prompt, how did you react?"
`Brave.Shields.CookieListPrompt`

0) Have not seen the prompt
1) Have seen the prompt, but did not react
2) Clicked "No thanks" on the prompt
3) Clicked "Block cookie notices" on the prompt

"Do you have cookie consent notice blocking enabled?"
`Brave.Shields.CookieListEnabled`

0) No
1) Yes

**Comment by kjozwiak** [comment]
The above requires `1.45.104` or higher for `1.45.x` verification 👍

**Comment by kjozwiak** [comment]
Labelling as `QA/Blocked` as we'll need to wait for https://github.com/brave/brave-browser/issues/26078 to be fixed and uplifted into `1.45.x`. We'll also need to enable `BraveAdblockCookieListOptInStudy` on `BETA` and `Release` before we can verify the above as we'll be enabling this feature via Grffin on both `Desktop` & `Android`.

@Uni-verse @srirambv I can run through this as I have the most context/been working on this for a while.

**Comment by GeetaSarvadnya** [comment]
Verification PASSED on

```
Brave | 1.45.109 Chromium: 107.0.5304.62 (Official Build) (64-bit)
-- | --
Revision | 1eec40d3a5764881c92085aaee66d25075c159aa-refs/branch-heads/5304@{#942}
OS | Windows 10 Version 21H2 (Build 19044.2130)

```

- Verified the test plan from https://github.com/brave/brave-core/pull/15187#issue-1383168782


**Brave.Shields.CookieListPrompt**

<details>
<summary>Brave.Shields.CookieListPrompt = 0_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://local-state file 
3. Ensured that the metrics `Brave.Shields.CookieListPrompt` is NOT shown 

</details>

<details>
<summary>Brave.Shields.CookieListPrompt = 1_PASSED</summary>

<details>
<summary>Case 1: Click on  close button on cookie consent block prompt</summary>

1. Clean profile 1.45.109
4. Enable the flag "Show an opt-in bubble for the "Easylist-Cookie List" filter" via brave://flags
5. Relaunch the browser
6. Ensure that the `cookie consent block` prompt is shown
7. Click on the `x` or close button in the popup
8. Open brave://local-state file and ensured that the metrics response value for the metrics `Brave.Shields.CookieListPrompt` is  `1` 

![image](https://user-images.githubusercontent.com/38657976/197253597-4f3b90f6-dc95-4a4d-ade7-a587a0880297.png)


</details>

<details>
<summary>Case 2: Open a new NTP when cookies consent modal popup prompt</summary>

1. Clean profile 1.45.109
2. Enable the flag "Show an opt-in bubble for the "Easylist-Cookie List" filter" via brave://flags
4. Relaunch the browser
5. Ensure that the `cookie consent block` prompt is shown
6. Open a new NTP and ensure cookies consent popup is dismissed
7. Open brave://local-state file and ensured that the metrics response value for the metrics `Brave.Shields.CookieListPrompt` is  `1` 

![image](https://user-images.githubusercontent.com/38657976/197253653-68de500c-6b53-4795-913b-1a1043474ed6.png)

</details>

<details>
<summary>Case 3: Close the `cookie consent block` modal popup tab</summary>

1. Clean profile 1.45.109
2. Open any site in NTP
3. Enable the flag "Show an opt-in bubble for the "Easylist-Cookie List" filter" via brave://flags
4. Relaunch the browser
5. Ensure that the `cookie consent block` prompt is shown
6. Close the active tab of `cookie consent block` prompt
9. Open brave://local-state file and ensured that the metrics response value for the metrics `Brave.Shields.CookieListPrompt` is  `1` 

![image](https://user-images.githubusercontent.com/38657976/197253879-cb129c0d-8157-458c-abf1-6f53a5ad03b3.png)


</details>

<details>
<summary>Case 4: Switch the tab to other tab when cookies consent modal popup prompt</summary>

1. Clean profile 1.46.x
2. Open any site in NTP
3. Enable the flag "Show an opt-in bubble for the "Easylist-Cookie List" filter" via brave://flags
4. Relaunch the browser
5. Ensure that the `cookie consent block` prompt is shown
6. Switch the tab to a site ( opened in step 2)
8. Open brave://local-state file and ensured that the metrics response value for the metrics `Brave.Shields.CookieListPrompt` is  `1` 

![image](https://user-images.githubusercontent.com/38657976/197254414-a0a23ad2-7f55-410d-b69b-cfc441afee1d.png)


</details>


</details>

<details>
<summary>Brave.Shields.CookieListPrompt = 2_PASSED</summary>

- Confirmed that the response value for the metrics `Brave.Shields.CookieListPrompt` is  2 when user clicks on the "No thanks" button on the `cookie consent block` prompt

Example | Example
-- | --


</details>

<details>
<summary>Brave.Shields.CookieListPrompt = 3_PASSED</summary>

- Confirmed that the response value for the metrics `Brave.Shields.CookieListPrompt` is  3 when user clicks on the "Block cookie notices" button on the `cookie consent block` prompt

Example | Example
-- | --



</details>

**Brave.Shields.CookieListEnabled**

<details>
<summary>Brave.Shields.CookieListEnabled = 0_PASSED</summary>

<details>
<summary>Case 1: Have not enabled the flag "Easylist-Cookie List"_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://local-state file
3. Ensured that the metrics `Brave.Shields.CookieListEnabled` response value is `0`

Example | Example
-- | --


</details>

<details>
<summary>Case 2: Enabled the flag "Easylist-Cookie List" and clicked on Close button_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://flags and enable the flag `Show an opt-in bubble for the "Easylist-Cookie List" filter`
3. Relaunch the browser
4. Confirmed that the cookies consent prompt is shown
5. Click on close (x) button
6. Open brave://local-state file
8. Ensured that the metrics `Brave.Shields.CookieListEnabled` response value is `0`  and `Brave.Shields.CookieListPrompt` response value is `1`

</details>

<details>
<summary>Case 3: Enabled the flag "Easylist-Cookie List" and closed the cookies consent tab_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://local-state
3. Open brave://flags and enable the flag `Show an opt-in bubble for the "Easylist-Cookie List" filter`
4. Relaunch the browser
5. Confirmed that the cookies consent prompt is shown
6. Close the cookies consent tab
7. Reload brave://local-state file
10. Ensured that the metrics `Brave.Shields.CookieListEnabled` response value is `0`  and `Brave.Shields.CookieListPrompt` response value is `1`

</details>

<details>
<summary>Case 4: Enabled the flag "Easylist-Cookie List" and switch to a tab other than the cookies consent tab_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://local-state
3. Open brave://flags and enable the flag `Show an opt-in bubble for the "Easylist-Cookie List" filter`
4. Relaunch the browser
5. Confirmed that the cookies consent prompt is shown
6. Switch to a tab other than the cookies consent tab
7. Reload brave://local-state file
9. Ensured that the metrics `Brave.Shields.CookieListEnabled` response value is `0`  and `Brave.Shields.CookieListPrompt` response value is `1`

</details>

</details>

<details>
<summary>Brave.Shields.CookieListEnabled = 1_PASSED</summary>

1. Clean profile 1.46.x
2. Open brave://flags and enable the flag `Show an opt-in bubble for the "Easylist-Cookie List" filter`
3. Relaunch the browser
4. Confirmed that the cookies consent prompt is shown
5. Click on the "Block cookie notices" on the prompt
7. Reload brave://local-state file
8. Confirmed that the metrics `Brave.Shields.CookieListEnabled` response value is `1` when user clicks on the "Block cookie notices"on the prompt and the metrics `Brave.Shields.CookieListPrompt` response value is `3`
11. Confirmed that the `Easylist - Cookie List` is enabled in brave://settings/shields/filters when user clicks on the "Block cookie notices" on the prompt

Example | Example | Example
-- | -- | --


</details>
